pve2_api.class.php

```diff
@@ -27,21 +27,21 @@
 
 class PVE2_Exception extends RuntimeException {}
 
+
 class PVE2_API {
 	protected $hostname;
 	protected $username;
 	protected $realm;
 	protected $password;
 	protected $port;
 	protected $verify_ssl;
-
 	protected $login_ticket = null;
 	protected $login_ticket_timestamp = null;
 	protected $cluster_node_list = null;
 
-	public function __construct ($hostname, $username, $realm, $password, $port = 8006, $verify_ssl = false) {
-		if (empty($hostname) || empty($username) || empty($realm) || empty($password) || empty($port)) {
-			throw new PVE2_Exception("Hostname/Username/Realm/Password/Port required for PVE2_API object constructor.", 1);
+	public function __construct ($hostname, $username, $realm, $password, $tokenid, $tokensecret, $port = 8006, $verify_ssl = false) {
```

Comment:
the way that `$tokenid` and `$tokensecret` have been added here - does this make it not backwards compatible...? ie. this would break people's code that relies on this library already?

would it make more sense to make these fields optional with empty defaults instead? I'm not super comfortable with breaking peoples existing code here...

Reply:
You're right, I didn't think this through 100%. The issue only happens if port and or verify_ssl are set, otherwise it would work.

Because we can't provide identifiers for the parameters like in python, if you'd want to use tokens, you'd need to specify port & verify_ssl (or provide null)

`$pmx = new PVE2_API($hostname, $username, $realm, $password, null, null, $tokenid, $tokensecret);`

I probably didn't consider it this way because "It doesn't look nice".